python/examples/feature-selection.py

```diff
@@ -0,0 +1,109 @@
+import sys
+
+import chemfiles
+
+from rascaline import SoapPowerSpectrum
+from equistore import Labels, TensorMap, TensorBlock
+import numpy as np
+from skcosmo.feature_selection import FPS
+
+with chemfiles.Trajectory(sys.argv[1]) as trajectory:
+    frames = [f for f in trajectory]
+
+HYPER_PARAMETERS = {
+    "cutoff": 5.0,
+    "max_radial": 6,
+    "max_angular": 4,
+    "atomic_gaussian_width": 0.3,
+    "center_atom_weight": 1.0,
+    "radial_basis": {
+        "Gto": {},
+    },
+    "cutoff_function": {
+        "ShiftedCosine": {"width": 0.5},
+    },
+}
+
+calculator = SoapPowerSpectrum(**HYPER_PARAMETERS)
+
+descriptor = calculator.compute(frames, gradients=["positions"])
+
+# The names of the properties that will be chosen during training
+# coincide with the properties of the blocks obtained after calculating the
+# properties. Accordingly, the names of the samples can be obtained with the
+# following commands (here it will be "l", "n1", "n2").
+descriptor.block(0).properties.names
+
+# selected_samples can be a variable of `Labels` or `TensorMap` type.
+# In the first case, the command will look like this (if we want to
+# calculate only features with l=0)
+labels = Labels(
+    names=["l"],
+    values=np.array([[0]]),
+)
+reference_descriptor = calculator.compute(frames, selected_properties=labels)
+
+# In the latter case, from the `TensorMap`, which we pass, the list
+# of properties is taken, which will be added to the final result. The
+# code with looks as following:
+
+descriptor_for_comparison = calculator.compute(
+    frames, selected_properties=reference_descriptor
+)
+print(f"shape of first block initially: {descriptor.block(0).values.shape}")
+print(
+    f"shape of first block of reference: {reference_descriptor.block(0).values.shape}"
+)
+print(
+    f"shape of first block after selection: {descriptor_for_comparison.block(0).values.shape}"
+)
+
+
+# Thanks to the TensorMap format, we get a feature matrix block
+# by block. The TensorMap format allows us to select features within
+# each block, and then construct a general matrix of features. We
+# can select the most significant features using FPS, which selects
+# features based on the  distance between them. The following algorithm
+# selects the 10 most important features in each block, then constructs
+# a new descriptor based on them, and calculates the final matrix of
+# features for it.
+
+# choose features:
+features_list = []
+for i in range(len(descriptor)):
+    fps = FPS(n_to_select=10)
+    values = descriptor.block(i).values
```

Comment:
maybe the code doing the selection should be moved to a separate function, to make the whole file easier to follow?

Reply:
> maybe the code doing the selection should be moved to a separate function, to make the whole file easier to follow?

I think it is worth explicitly showing exactly how the FPS for each block occurs. If people need to implement it themselves, there's no need to hide parts of the code from them, as I think.

Reply:
I don't mean a hidden function, but defining a function inside this example file! This way, people are also able to copy/paste and re-use this function =)

Reply:
> I don't mean a hidden function, but defining a function inside this example file! This way, people are also able to copy/paste and re-use this function =)

Ah, okay, I agree!